Add scripts that parse and compare static analyzer HTML output

briannafan opened the pull request:
#### 49272b1cbf82a0302b27f54f400ec8275bbaacc3
<pre>
Add scripts that parse and compare static analyzer HTML output
<a href="https://bugs.webkit.org/show_bug.cgi?id=269390">https://bugs.webkit.org/show_bug.cgi?id=269390</a>
<a href="https://rdar.apple.com/problem/122962791">rdar://problem/122962791</a>

Reviewed by David Kilzer.

These scripts will be used by a static analyzer bot to parse and compare HTML reports from static analysis.

* Tools/Scripts/compare-static-analysis-results.py: Added.
(parser):
(find_diff): Compares two files to find regressions and fixes.
(compare_project_results): Consolidates results per project.
(create_filtered_results_dir): Uses scan-build --generate-index-only to generate new index.html for new issues per project.
(main):
* Tools/Scripts/generate-dirty-files.py: Added. Takes in output from the static analyzer and extracts
the issue hash, file name, and line number per project and checker type.
(parser):
(parse_results_file): Extracts data from HTML comments.
(find_project_results): Writes data per checker.
(find_all_results): Summarizes all results.
(main):

Canonical link: <a href="https://commits.webkit.org/276495@main">https://commits.webkit.org/276495@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/079c877ec2856873aa08ddf67150f824938983f2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/44507 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/23584 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46956 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/47159 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/40515 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/46812 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/27595 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20974 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/36605 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/45083 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/20632 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/38307 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17665 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/18079 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/39447 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/2556 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40695 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39724 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48779 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/19470 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/16004 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43530 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/20817 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/42273 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9982 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/21145 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/20466 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer reviewed (approved):
r=me.  We should land this so we can continue to iterate.  (Sorry I didn't see this much earlier!)

briannafan committed:
Add scripts that parse and compare static analyzer HTML output
https://bugs.webkit.org/show_bug.cgi?id=269390
rdar://problem/122962791

Reviewed by David Kilzer.

These scripts will be used by a static analyzer bot to parse and compare HTML reports from static analysis.

* Tools/Scripts/compare-static-analysis-results.py: Added.
(parser):
(find_diff): Compares two files to find regressions and fixes.
(compare_project_results): Consolidates results per project.
(create_filtered_results_dir): Uses scan-build --generate-index-only to generate new index.html for new issues per project.
(main):
* Tools/Scripts/generate-dirty-files.py: Added. Takes in output from the static analyzer and extracts
the issue hash, file name, and line number per project and checker type.
(parser):
(parse_results_file): Extracts data from HTML comments.
(find_project_results): Writes data per checker.
(find_all_results): Summarizes all results.
(main):

Canonical link: https://commits.webkit.org/276495@main
